src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleArray.cs

```diff
@@ -162,6 +162,11 @@ public static partial class JsonSerializer
                 }
                 else
                 {
+                    if (state.Current.ReturnValue == null)
+                    {
+                        ThrowHelper.ThrowJsonException_DeserializeUnableToConvertValue(value.GetType(), reader, state.JsonPath);
+                        return;
+                    }
```

Comment:
Do we need to guard against the potential that ReturnValue is non-null but not an IList? Even if we don't, I suspect it would be a tad cheaper to do:
```C#
if (!(state.Current.ReturnValue is IList list))
{
    ThrowHelper.Throw...;
    return;
}
list.Add(value);
```
rather than dereferencing `state.Current.ReturnValue` twice.